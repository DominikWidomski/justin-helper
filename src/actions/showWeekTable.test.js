const bufferConsoleLog = require("./utils/consoleBuffer");
const showWeekTable = require("./showWeekTable");

describe("actions", () => {    
    describe("showWeekTable", () => {
        test("default", () => {
            // There's a thing with Jest where I can customise the snapshot output right???
            // Check out the snapshot diffs thing
            expect(bufferConsoleLog(showWeekTable)).toMatchSnapshot();
        });

        test("single project", () => {
            const projectTimes = [{
                id: 1,
                attributes: {
                    user_id: '1',
                    project_id: '1',
                    date: '2018-01-01',
                    duration_mins: '420',
                    is_rejected: 'false'
                },
            }];

            const projects = {
                data: [{
                    id: '1',
                    attributes: {
                        name: 'project',
                    }
                }]
            };

            const options = {
                startDate: '2018-01-01',
                endDate: '2018-01-02'
            };

            expect(bufferConsoleLog(showWeekTable, projectTimes, projects, options)).toMatchSnapshot();
        })

        test("multiple times in a day", () => {
            const projectTimes = [{
                id: 1,
                attributes: {
                    user_id: '1',
                    project_id: '1',
                    date: '2018-01-01',
                    duration_mins: '300',
                    is_rejected: 'false'
                },
            }, {
                id: 2,
                attributes: {
                    user_id: '1',
                    project_id: '2',
                    date: '2018-01-01',
                    duration_mins: '120',
                    is_rejected: 'false'
                },
            }];

            const projects = {
                data: [{
                    id: '1',
                    attributes: {
                        name: 'project',
                    }
                }, {
                    id: '2',
                    attributes: {
                        name: 'project 2',
                    }
                }]
            };

            const options = {
                startDate: '2018-01-01',
                endDate: '2018-01-02'
            };

            expect(bufferConsoleLog(showWeekTable, projectTimes, projects, options)).toMatchSnapshot();
        })
    })
})