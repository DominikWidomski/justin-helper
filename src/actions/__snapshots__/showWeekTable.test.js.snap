// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions showWeekTable default 1`] = `
"
┌─────┬──────┬─────────┬──────┐
│ Day │ Date │ Project │ Time │
└─────┴──────┴─────────┴──────┘
"
`;

exports[`actions showWeekTable multiple times in a day 1`] = `
"
┌────────┬────────────┬───────────┬──────┐
│ Day    │ Date       │ Project   │ Time │
├────────┼────────────┼───────────┼──────┤
│ Monday │ 2018-01-01 │ project   │ 5    │
├────────┼────────────┼───────────┼──────┤
│        │            │ project 2 │ 2    │
└────────┴────────────┴───────────┴──────┘
"
`;

exports[`actions showWeekTable single project 1`] = `
"
┌────────┬────────────┬─────────┬──────┐
│ Day    │ Date       │ Project │ Time │
├────────┼────────────┼─────────┼──────┤
│ Monday │ 2018-01-01 │ project │ 7    │
└────────┴────────────┴─────────┴──────┘
"
`;
